packages/ui/src/elements/WhereBuilder/Condition/index.tsx

```diff
@@ -103,6 +104,14 @@ export const Condition: React.FC<Props> = (props) => {
 
   const handleOperatorChange = useCallback(
     (operator: Option<Operator>) => {
+      const validOperatorValue = operatorValueTypes[operator.value] || 'any'
+      const isValidValue = validOperatorValue === 'any' || typeof value === validOperatorValue
```

Comment:
Are numeric values actually numbers here or are they strings? I just want to make sure the typeof check will work properly for them.

Reply:
> Are numeric values actually numbers here or are they strings? I just want to make sure the typeof check will work properly for them.

This sounds legit. Need to look into it.

Reply:
@JessChowdhury was it an object because `value` was null? I would assume that is the only case where that would be true.

Reply:
@JarrodMFlesch @jacobsfletch 

I dug a little deeper and found:

- The operators **[greater than, less than, greater or equal, less or equal]** appear for number fields and date fields. 
  - If it’s a number field, the value and type is a `string`. 
  - If it is a date field the value looks like `Mon Feb 17 2025 12:00:00 GMT+0000 (Greenwich Mean Time)` and the type is `object` 

The only other anomaly I found is with `exists` - this gives `true / false` but the value type is `string` instead of `boolean`

So I may need to change the approach since the "valid operator type" changes based on the field in some cases.